pkg/collector/ecr_events_test.go

```diff
@@ -0,0 +1,245 @@
+				testPackageDetails["MINIMAL"],
+				testPackageDetails["LOW"],
+				testPackageDetails["MEDIUM"],
+				testPackageDetails["HIGH"],
+				testPackageDetails["CRITICAL"],
+			},
+		},
+	},
+}
+
+type ECRTestSuite struct {
+	suite.Suite
+}
+
+type mockECRClient struct{}
+
+func (m *mockECRClient) DescribeImageScanFindings(input *ecr.DescribeImageScanFindingsInput) (*ecr.DescribeImageScanFindingsOutput, error) {
+	if _, ok := testFindingsOutput[*input.ImageId.ImageDigest]; ok {
+		return testFindingsOutput[*input.ImageId.ImageDigest], nil
+    }
+
+    return nil, errors.New("error")
+}
+
+func (suite *ECRTestSuite) TestGetNoVulnerabilityDetail() {
+	client := &mockECRClient{}
+
+	testECRDetail := &ECRImageScanDetail{
+		ScanStatus:     ecr.ScanStatusComplete,
+		RepositoryName: "testRepo",
+		ImageDigest:    "zeroVulnerabilities",
+	}
+
+	vulnerabilityOccurences, _ := getVulnerabilityDetails(client, testECRDetail)
+
+	assert.Len(suite.T(), vulnerabilityOccurences, 0)
+}
+
+func (suite *ECRTestSuite) TestGetOneVulnerabilityDetail() {
+	client := &mockECRClient{}
+
+	testECRDetail := &ECRImageScanDetail{
+		ScanStatus:     ecr.ScanStatusComplete,
+		RepositoryName: "testRepo",
+		ImageDigest:    "singleVulnerability",
+	}
+
+	result, _ := getVulnerabilityDetails(client, testECRDetail)
+
+	assert.Len(suite.T(), result, 1)
+	assert.Equal(suite.T(), result[0].Vulnerability.Severity.String(), *testPackageDetails["CRITICAL"].Severity)
+	assert.Equal(suite.T(), result[0].Vulnerability.PackageIssue[0].AffectedLocation.CpeUri, *testPackageDetails["CRITICAL"].Uri)
+	assert.Equal(suite.T(), result[0].Vulnerability.PackageIssue[0].AffectedLocation.Version.Name, *testPackageDetails["CRITICAL"].Attributes[0].Value)
+	assert.Equal(suite.T(), result[0].Vulnerability.PackageIssue[0].AffectedLocation.Package, *testPackageDetails["CRITICAL"].Attributes[1].Value)
+}
+
+func (suite *ECRTestSuite) TestGetMultipleVulnerabilityDetail() {
+	client := &mockECRClient{}
+
+	testECRDetail := &ECRImageScanDetail{
+		ScanStatus:     ecr.ScanStatusComplete,
+		RepositoryName: "testRepo",
+		ImageDigest:    "manyVulnerabilities",
+	}
+
+	result, _ := getVulnerabilityDetails(client, testECRDetail)
+
+	assert.Len(suite.T(), result, 6)
+	for i := range result {
+		s := result[i].Vulnerability.Severity.String()
+		assert.Equal(suite.T(), result[i].Vulnerability.PackageIssue[0].AffectedLocation.CpeUri, *testPackageDetails[s].Uri)
+		assert.Equal(suite.T(), result[i].Vulnerability.PackageIssue[0].AffectedLocation.Version.Name, *testPackageDetails[s].Attributes[0].Value)
+	}
+}
+
+func (suite *ECRTestSuite) TestBadInput() {
+	client := &mockECRClient{}
+
+	testECRDetail := &ECRImageScanDetail{
+		ScanStatus:     ecr.ScanStatusComplete,
+		RepositoryName: "testRepo",
+		ImageDigest:    "nonExistentImage",
+	}
+
+  _, err := getVulnerabilityDetails(client, testECRDetail)
+
+  assert.Error(suite.T(), err)
+}
+func TestECRTestSuite(t *testing.T) {
+	suite.Run(t, new(ECRTestSuite))
+}
```

Comment:
love the tests!  my only thinking is that maybe we'd want to standardize on ginkgo / gomega for our tests, since imo they look much cleaner.  but there's already other testify based tests in here that I imagine you referenced, so we don't have to do that quite yet.

Reply:
Yup, I had seen testify getting used in other places so I was trying to keep the tooling consistent. I have no problem with switching to ginkgo/gomega going forward but like you said that's probably another piece of work.